### DISCUSSION TITLE
WiX v4 tutorial

### DISCUSSION THREAD

**Discussion by Sanqui** [wixtoolset/issues#7576] [Questions]
Apologies for the basic question, but since this is a discussion board I figure it to be acceptable.
Since the stable release of WiX Toolset is now v4, is a tutorial ready for beginners of the framework?  Most resources I can find, including the [Firegiant WiX Toolset Tutorial](https://www.firegiant.com/wix/tutorial/), are geared towards v3, or talk about changes from v3 to v4, but I'm unable to find any describing how to start a project using v4.  There is [documentation for v4](https://wixtoolset.org/docs/intro/), but it jumps straight into concepts like ["Burn bundles"](https://wixtoolset.org/docs/tools/burn/) without any introduction.

Is there a recommended tutorial to WiX v4 that I have missed, or are there plans to update the documentation with some introduction?  Should I go ahead and dive into the v3 tutorial and then read about migration?  Thanks!

**Comment by wmanning** [comment]
One of the wix creators Rob did a video series on v4 (Deployment Dojo): https://www.youtube.com/watch?v=-Y7dOtciOJw&list=PLDlzbQXIs18slmqmdlS10_de_Cps-QRg6

**Reply by robmen** [reply]
The Deployment Dojo is an ongoing series. In fact the next live episode starts in just 3 hours. You can watch live or catch the recording on the play list shared above.

Also, we are working on the v4 tutorial at FireGiant. It takes a good bit of effort to update 100ish pages of documentation. We also prioritize the desires of customers at FireGiant so the more customers interested in the v4 tutorial, the more effort we put into it.

**Reply by BryanCrotaz** [reply]
15 minutes into the "tutorial" at that link and he's told me all the things that *won't* work. Is there not a simple one page 5 step tutorial anywhere to get a simple installer that installs one file from one project output? The other 100s of pages are only useful once you know how the concepts work together.

**Reply by robmen** [reply]
I don't know which "link" you are referring to. The Deployment Dojo isn't a 15-minute tutorial. None of the episodes are 15 minutes, so I'm guessing you aren't referring to the link to the video series. 😄 

I didn't see a direct link to [FireGiant's WiX v4 tutorial](https://www.firegiant.com/docs/wix/tutorial/) so maybe that's more what you want? It isn't a "one page 5 step tutorial" because it assumes the reader has very little context and starts by providing a foundation for all future topics to build upon.

**Comment by robmen** [comment]
At FireGiant, we are creating a [WiX v4 tutorial](https://www.firegiant.com/docs/wix/tutorial/) to modernize our v3 tutorial. There is still a list of topics to write on the backlog, but it already provides a very good foundation for getting started.

**Reply by kkarasti** [reply]
I used to use the VS Installer Projects extension but it seems broken now after I updated my VS to the latest version. So I figured it was time to look at something else. I added the HeatWave extension to VS, followed the v4 tutorial and built the setup package. I run the msi and yes, my app is in the ProgramFiles directory. I double click the exe and the app doesn't run, nothing. The app runs fine in debug in VS. The app is a WPF app based on NET6.

Everything I've read makes it sound like this should work. What am I doing wrong?

**Reply by robmen** [reply]
You don't provide enough information to know for sure but my first guess is that you didn't install all of the files of your app.

**Reply by kkarasti** [reply]
I made a new project following the tutorial, this time I targeted NET4.7.2, and it works fine. So does Wix not work with NET6+ ??
I'm a total noob to Wix and was hoping the tutorial would get me started. All the info I find out there either doesn't work or targets old frameworks.

**Reply by robmen** [reply]
The WiX Toolset can create an installation package that installs anything. Putting the correct stuff in your installation package is up to you. I bet you are not including all the files needed to run your .NET 6 application in your installation package. You don't provide any information, so it is hard to say for sure.

**Reply by kkarasti** [reply]
I don't know what further info you want. Follow the tutorial, make a WinForms app, target .NET6 (instead of 4.7.2) and it doesn't work. And that's the problem, don't know what to do net. Try it yourself.

**Reply by robmen** [reply]
If that is the sole problem, I believe I gave you the answer: 

>  I bet you are not including all the files needed to run your .NET 6 application in your installation package.

I don't usually share screen-shots, but visually, you can see a difference between .NET Framework WinForms and .NET 6 WinForms.

.NET Framwork output looks like this:

![image](https://github.com/wixtoolset/issues/assets/131904/38f596d5-4a57-4dc0-af16-c9d4b26338da)

.NET 6 WinForms looks like this:

![image](https://github.com/wixtoolset/issues/assets/131904/72de298a-d063-48d4-975e-33f2d5afc76f)

You don't usually deploy the `.pdb` file and the `.exe.config` file _can_ be optional. But the other files are vital. Did you add the extra files for .NET 6? If not, then your .NET 6 app won't run.

**Reply by kkarasti** [reply]
Ahhh, I get it now. A different way of thinking. Thanks.
